src/language/JSUtils.js

```diff
@@ -68,19 +57,99 @@ define(function (require, exports, module) {
     function _findAllFunctionsInText(text) {
         var results = {},
             functionName,
+            resultNode,
+            memberPrefix,
             match;
-
+   
         PerfUtils.markStart(PerfUtils.JSUTILS_REGEXP);
-
-        while ((match = _functionRegExp.exec(text)) !== null) {
-            functionName = (match[2] || match[5]).trim();
-
+        
+        var AST = AcornLoose.parse_dammit(text, {locations: true});
+        
+        function _addResult(node, offset, prefix) {
+            memberPrefix = prefix ? prefix + " - " : "";
+            resultNode = node.id || node.key || node;
+            functionName = resultNode.name;
             if (!Array.isArray(results[functionName])) {
                 results[functionName] = [];
             }
 
-            results[functionName].push({offsetStart: match.index});
+            results[functionName].push(
+                {
+                    offsetStart: offset || node.start,
+                    label: memberPrefix ? memberPrefix + functionName : null,
+                    location: resultNode.loc
+                }
+            );
         }
+        
+        ASTWalker.simple(AST, {
+            /*
+                function <functionName> () {}
+            */
+            FunctionDeclaration: function (node) {
+                if (node.id.name !== '✖') { // A hack to reject anonymous declarations 
```

Comment:
I am sorry @ficristo , misinterpreted the comment 😞 . I have changed the comment in the code now  and also added the ArrowFunctionExpression construct along with addressing other code review comments and test cases.